Mathlib/Topology/Category/CompHaus/ExplicitLimits.lean

```diff
@@ -0,0 +1,177 @@
+  CompHaus.of set
+
+/--
+The projection from the pullback to the first component.
+-/
+def pullback.fst : pullback f g ⟶ X where
+  toFun := fun ⟨⟨x,_⟩,_⟩ => x
+  continuous_toFun := Continuous.comp continuous_fst continuous_subtype_val
+
+/--
+The projection from the pullback to the second component.
+-/
+def pullback.snd : pullback f g ⟶ Y where
+  toFun := fun ⟨⟨_,y⟩,_⟩ => y
+  continuous_toFun := Continuous.comp continuous_snd continuous_subtype_val
+
+@[reassoc]
+lemma pullback.condition : pullback.fst f g ≫ f = pullback.snd f g ≫ g := by
+  ext ⟨_,h⟩ ; exact h
+
+/--
+Construct a morphism to the explicit pullback given morphisms to the factors
+which are compatible with the maps to the base.
+This is essentially the universal property of the pullback.
+-/
+def pullback.lift {Z : CompHaus.{u}} (a : Z ⟶ X) (b : Z ⟶ Y) (w : a ≫ f = b ≫ g) :
+    Z ⟶ pullback f g where
+  toFun := fun z => ⟨⟨a z, b z⟩, by apply_fun (fun q => q z) at w ; exact w⟩
+  continuous_toFun := by
+    apply Continuous.subtype_mk
+    rw [continuous_prod_mk]
+    exact ⟨a.continuous, b.continuous⟩
+
+@[reassoc (attr := simp)]
+lemma pullback.lift_fst {Z : CompHaus.{u}} (a : Z ⟶ X) (b : Z ⟶ Y) (w : a ≫ f = b ≫ g) :
+  pullback.lift f g a b w ≫ pullback.fst f g = a := rfl
+
+@[reassoc (attr := simp)]
+lemma pullback.lift_snd {Z : CompHaus.{u}} (a : Z ⟶ X) (b : Z ⟶ Y) (w : a ≫ f = b ≫ g) :
+  pullback.lift f g a b w ≫ pullback.snd f g = b := rfl
+
+lemma pullback.hom_ext {Z : CompHaus.{u}} (a b : Z ⟶ pullback f g)
+    (hfst : a ≫ pullback.fst f g = b ≫ pullback.fst f g)
+    (hsnd : a ≫ pullback.snd f g = b ≫ pullback.snd f g) : a = b := by
+  ext z
+  apply_fun (fun q => q z) at hfst hsnd
+  apply Subtype.ext
+  apply Prod.ext
+  · exact hfst
+  · exact hsnd
+
+/--
+The pullback cone whose cone point is the explicit pullback.
+-/
+@[simps! pt π]
+def pullback.cone : Limits.PullbackCone f g:=
+  Limits.PullbackCone.mk (pullback.fst f g) (pullback.snd f g) (pullback.condition f g)
+
+/--
+The explicit pullback cone is a limit cone.
+-/
+@[simps! lift]
+def pullback.isLimit : Limits.IsLimit (pullback.cone f g) :=
+  Limits.PullbackCone.isLimitAux _
+    (fun s => pullback.lift f g s.fst s.snd s.condition)
+    (fun _ => pullback.lift_fst _ _ _ _ _)
+    (fun _ => pullback.lift_snd _ _ _ _ _)
+    (fun _ _ hm => pullback.hom_ext _ _ _ _ (hm .left) (hm .right))
+
+end Pullbacks
+
+section FiniteCoproducts
+
+variable {α : Type} [Fintype α] (X : α → CompHaus.{u})
+
+/--
+The coproduct of a finite family of objects in `CompHaus`, constructed as the disjoint
+union with its usual topology.
+-/
+def finiteCoproduct : CompHaus := CompHaus.of <| Σ (a : α), X a
+
+/--
+The inclusion of one of the factors into the explicit finite coproduct.
+-/
+def finiteCoproduct.incl (a : α) : X a ⟶ finiteCoproduct X where
+  toFun := fun x => ⟨a,x⟩
+  continuous_toFun := continuous_sigmaMk (σ := fun a => X a)
```

Comment:
Maybe still name this `\iota` to match the limits library? Not essential.

Reply:
Done!